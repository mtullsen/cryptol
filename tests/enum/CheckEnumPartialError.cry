enum Maybe a = Nothing | Just a

maybeMapBad1 : {a, b} (a -> b) -> Maybe a -> Maybe b
maybeMapBad1 f m =
  case m of
    Just x -> Just (f x)

maybeMapBad2 : {a, b} (a -> b) -> Maybe a -> Maybe b
maybeMapBad2 f m =
  case m of
    Nothing -> Nothing

enum Foo = A | B | C

fooBad1 : Foo -> ()
fooBad1 f =
  case f of
    A -> ()

fooGood : Foo -> ()
fooGood f =
  case f of
    A -> ()
    _ -> ()
