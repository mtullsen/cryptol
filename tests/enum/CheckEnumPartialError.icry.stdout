Loading module Cryptol
Loading module Cryptol
Loading module Main

[error] at CheckEnumPartialError.cry:5:3--8:1:
  Case expression does not cover the following patterns:
  Main::Nothing
[error] at CheckEnumPartialError.cry:10:3--13:1:
  Case expression does not cover the following patterns:
  Main::Just
[error] at CheckEnumPartialError.cry:17:3--20:1:
  Case expression does not cover the following patterns:
  Main::B, Main::C
